tests/drivers/build_all/modem/modem_cellular.conf

```diff
@@ -0,0 +1,8 @@
+CONFIG_TEST=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_NETWORKING=y
+CONFIG_NET_L2_PPP=y
+CONFIG_MODEM=y
+CONFIG_MODEM_CELLULAR=y
```

Comment:
needs `CONFIG_MODEM_MODULES=y` right?
I tried building the test and:
```
ryan@superfruit modem % west build -b pinnacle_100_dvk -- -DCONF_FILE=modem_cellular.conf
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /Users/ryan/git/zephyr_workspace/zephyr/tests/drivers/build_all/modem
-- CMake version: 3.26.4
-- Found Python3: /Users/ryan/.pyenv/versions/3.10.5/bin/python3.10 (found suitable exact version "3.10.5") found components: Interpreter 
-- Cache files will be written to: /Users/ryan/Library/Caches/zephyr
-- Zephyr version: 3.3.99 (/Users/ryan/git/zephyr_workspace/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
-- Board: pinnacle_100_dvk
-- Found host-tools: zephyr 0.15.0 (/Users/ryan/zephyr-sdk-0.15.0)
-- Found toolchain: zephyr 0.15.0 (/Users/ryan/zephyr-sdk-0.15.0)
-- Found Dtc: /usr/local/bin/dtc (found suitable version "1.7.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /Users/ryan/git/zephyr_workspace/zephyr/boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts
-- Found devicetree overlay: /Users/ryan/git/zephyr_workspace/zephyr/tests/drivers/build_all/modem/app.overlay
-- Generated zephyr.dts: /Users/ryan/git/zephyr_workspace/zephyr/tests/drivers/build_all/modem/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /Users/ryan/git/zephyr_workspace/zephyr/tests/drivers/build_all/modem/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /Users/ryan/git/zephyr_workspace/zephyr/tests/drivers/build_all/modem/build/zephyr/dts.cmake

warning: MODEM_CELLULAR (defined at drivers/modem/Kconfig.cellular:4) was assigned the value 'y' but
got the value 'n'. Check these unsatisfied dependencies: MODEM_MODULES (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_MODEM_CELLULAR and/or look up
MODEM_CELLULAR in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.
```

Reply:
CONFIG_MODEM_CELLULAR is the driver, which is only enabled if you have the CONFIG_MODEM enabled. A cheatsheet can be found in the build_all for the modem driver :) The driver will select the CONFIG_MODEM_MODULES

```
CONFIG_SERIAL=y
CONFIG_UART_INTERRUPT_DRIVEN=y
CONFIG_NETWORKING=y
CONFIG_NET_L2_PPP=y
CONFIG_MODEM=y
CONFIG_MODEM_CELLULAR=y
```

We may want to merge MODEM and MODEM_MODULES into one :)

Reply:
Note that this driver doesn't have an API like gsm_ppp_start(), it uses PM_DEVICE for power management

Reply:
I had to add `CONFIG_MODEM_MODULES=y` to `tests/drivers/build_all/modem/modem_cellular.conf` to get the test to build the driver. Otherwise the above warning is printed. It is not selecting the modules automatically.

Reply:
Noted.

Do you think we should move the sources and headers to `drivers/modem`?

I initially moved all new modules into their own folder to avoid naming conflicts, but I see now there are none.

Then we can replace CONFIG_MODEM_MODULES with CONFIG_MODEM

Reply:
put it all under `drivers/modem` and no `subsys/modem`? Yeah, that may make sense so its all in one spot.